notify/impl.go

```diff
@@ -1181,36 +1202,36 @@ func (n *VictorOps) Notify(ctx context.Context, as ...*types.Alert) (bool, error
 		level.Debug(n.logger).Log("msg", "Truncated stateMessage due to VictorOps stateMessage limit", "truncated_state_message", stateMessage, "incident", key)
 	}
 
-	msg := &victorOpsMessage{
-		MessageType:       messageType,
-		EntityID:          hashKey(key),
-		EntityDisplayName: tmpl(n.conf.EntityDisplayName),
-		StateMessage:      stateMessage,
-		MonitoringTool:    tmpl(n.conf.MonitoringTool),
+	msg := map[string]string{
+		"message_type":        messageType,
+		"entity_id":           hashKey(key),
+		"entity_display_name": tmpl(n.conf.EntityDisplayName),
+		"state_message":       stateMessage,
+		"monitoring_tool":     tmpl(n.conf.MonitoringTool),
 	}
 
 	if err != nil {
-		return false, fmt.Errorf("templating error: %s", err)
+		return nil, fmt.Errorf("templating error: %s", err)
 	}
 
-	var buf bytes.Buffer
-	if err := json.NewEncoder(&buf).Encode(msg); err != nil {
-		return false, err
-	}
+	for k, v := range n.conf.CustomFields {
 
-	c, err := commoncfg.NewClientFromConfig(*n.conf.HTTPConfig, "victorops")
-	if err != nil {
-		return false, err
+		// Validate if the custom field is not one of the fixed fields above.
```

Comment:
I think this would be better as a config-time error to the user, rather than silently dropping the data.

Reply:
I was wondering about this, it did feel a bit more like it should be a config time error, wasn't sure.

Reply:
Now that you validate fields in the configuration, this check can be removed.